### PR TITLE
Disable spell of word under cursor while in insertmode

### LINKED ISSUE #59: Don't highlight in insert mode over word
So I don't know if this is really possible to achieve, but I wanted to open an issue about it to hear if it is possible to do.

With the built-in spellchecker, it is not highlighted while in insertmode and typing out the word. See video below

https://user-images.githubusercontent.com/5160701/152340714-63fed0c5-5e7e-42e9-b95b-de785ee3f89a.mov

Here is the same thing with `spellsitter.nvim` enabled:

https://user-images.githubusercontent.com/5160701/152340787-2d7a394f-8a20-4913-8faf-2e922e2d3f70.mov

Is it possible to make it not highlight while the cursor is over the word and in insert mode? It's a bit distracting to see it highlight the word as bad spelling while typing out the word

### PR DESCRIPTION
**Pull request by seblj**
Closes #59

### CODE REVIEW

**Comment by seblj** [comment]
@lewis6991

**Comment by lewis6991** [comment]
Sorry. This fell off my radar. Will keep this notification in my inbox and get to this when I can. I want to test a thing or two regarding it.

File: lua/spellsitter.lua [hunk 1]
```diff
@@ -72,6 +72,11 @@ local function add_extmark(bufnr, lnum, col, len, highlight)
   -- TODO: This errors because of an out of bounds column when inserting
   -- newlines. Wrapping in pcall hides the issue.
 
+  local cursor = vim.api.nvim_win_get_cursor(0)
```
Comment (lewis6991) [inline lua/spellsitter.lua:75]: This might be more readable:
```lua
local cur_lnum, cur_col = unpack(vim.api.nvim_win_get_cursor(0))
```

**Review by lewis6991 — CHANGES_REQUESTED** [review]
See comments

**Pull request merged**